packages/@lwc/synthetic-shadow/src/shared/static-html-collection.ts

```diff
@@ -47,76 +42,27 @@ StaticHTMLCollection.prototype = create(HTMLCollection.prototype, {
         enumerable: true,
         configurable: true,
         value(name: string) {
-            if (isValidHTMLCollectionName(name) && this[name]) {
-                return this[name];
+            if (name === '') {
+                return null;
             }
-            const items = getHiddenField(this, Items);
+
+            const items = getHiddenField(this, Items) as Element[];
```

Comment:
another bug in the existing code is to assume that the `this` value does have the hidden field, which it might not, maybe we can fix this now.

Reply:
This is a more wide-spread issue. Let's get that in and create another PR. This PR has already been opened for too long.

Reply:
Now that I think about it, ensuring that the `this` does have the hidden field sounds overkill. We don't globally patch `HTMLCollection` and `NodeList`, the constructors are only used to create instances on specific APIs (eg. `document.getElementsByTagName`).

Because of this, the only way that I see where we would run into a situation where the `this` value doesn't have the hidden field by extracting one of those methods out of the instance and calling it with a different `this` value:

```
const { namedItem } = document.getElementsByTagName('div');
namedItem.call({ }, '1');
```

What are the odds that somebody will do that?